nesta/packages/meetup/meetup_utils.py

```diff
@@ -59,13 +64,56 @@ def flatten_data(list_json_data, keys, **kwargs):
                 else:
                     field_name = "_".join(k)
                     # Recursively assign the list of keys
-                    value = info                    
+                    value = info
                     for _k in k:
                         value = value[_k]
             # Ignore fields which aren't found (these will appear
             # as NULL in the database anyway)
             except KeyError:
                 continue
-            row[field_name] = value        
+            row[field_name] = value
         output.append(row)
     return output
+
+
+def get_members_by_percentile(engine, perc=10):
+    """Get the number of meetup group members for a given percentile
+    from the database.
+    
+    Args:
+        engine: A SQL alchemy connectable.
+        perc (int): A percentile to evaluate.
+    Returns:
+        members (float): The number of members corresponding to this percentile.
+    """
+    with db_session(engine) as session:
+        rows = (session
+                .query(Group.members)
+                .all())
+        rows = [r[0] for r in rows]
+    return float(np.percentile(rows, perc))
+
+
+def get_core_topics(engine, core_categories, members_limit, perc=99):
+    """Get the most frequent topics from a selection of meetup categories,
+    from the database.
+    
+    Args:
+        engine: A SQL alchemy connectable.
+        core_categories (list): A list of category_shortnames.
+        members_limit (int): Minimum number of members required in a group 
+                             for it to be considered.
+        perc (int): A percentile to evaluate the most frequent topics.
+    Returns:
+        topics (set): The set of most frequent topics.
+    """
+    with db_session(engine) as session:
+        rows = (session
+                .query(Group.topics)
+                .filter(Group.members >= members_limit)
+                .filter(Group.category_shortname.in_(core_categories))
```

Comment:
I didn't realise you could combine multiple `.filter` like this for AND, I have been using ugly multi-line bitwise operations 👍

Reply:
functional programming ftw